release-20.1: builtins: fix sqrdiff when used as a window function

yuzefovich opened the pull request:
Backport 1/1 commits from #55957.
Backport 1/1 commits from #56006.

/cc @cockroachdb/release

---

**builtins: fix sqrdiff when used as a window function**

Fix incorrect sqrdiff evaluation due to reuse of the results from
previous iterations when used as a window function.

Fixes: #55944.

Release note (bug fix): CockroachDB previously could incorrectly
evaluate sqrdiff function when used as a window function in some
cases, and now it is fixed.

**logictest: deflake a test**

We've recently merged a test that in very rare circumstances could
produce a float result that differs from the expected one by 1 in the
15th significant digit (after rounding). I believe that could occur,
e.g. when the 15th and 16th significant digits were `35`, and we matched
the spec of supporting 15 significant digits for floats, yet the
rounding makes us return an unexpected result. This commit rounds to the
precision of 1 digit less which should make the test non-flaky.

Release note: None

mneverov committed:
builtins: fix sqrdiff when used as a window function

Fix incorrect sqrdiff evaluation due to reuse of the results from
previous iterations when used as a window function.

Release note (bug fix): CockroachDB previously could incorrectly
evaluate sqrdiff function when used as a window function in some
cases, and now it is fixed.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/55999)
<!-- Reviewable:end -->

yuzefovich committed:
logictest: deflake a test

We've recently merged a test that in very rare circumstances could
produce a float result that differs from the expected one by 1 in the
15th significant digit (after rounding). I believe that could occur,
e.g. when the 15th and 16th significant digits were `35`, and we matched
the spec of supporting 15 significant digits for floats, yet the
rounding makes us return an unexpected result. This commit rounds to the
precision of 1 digit less which should make the test non-flaky.

Release note: None

asubiotto reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> in principle minus the CI failures

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/55999#-:-MKdVmqf7KeMHe4K9ns5:b-oj6ecs)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @asubiotto)



<!-- Sent from Reviewable.io -->

yuzefovich commented:
It took quite a bit of updating the logic tests (`F` type is only supported on the master branch, so I had to round the numbers).
